lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -925,33 +933,40 @@ findMatchingConcatDimForSameSizeSubTensor(llvm::ArrayRef<size_t> firstDims,
 /// there is a dimension in the input tensors, with which we can meet
 /// two requirements:
 /// 1) Input tensors are concatenate-able along this dimension.
-/// 2) The sub-tensors after this dimension in the input tensors, are
-///    of the same size as the sub-tensor of the input of the original
-///    Concat node after the concatenation dimension.
+/// 2) The trailing/leading dimensions sizes after/before this dimension in the
+///    input tensors, are of the same size as the trailing/leading dimensions of
+///    the input of the original Concat node after/before the concatenation
+///    dimension.
 /// \returns this dimension if found, otherwise -1.
 static int tryToConcatAlongSameSizeSubTensor(llvm::ArrayRef<NodeValue> inputs,
```

Comment:
I'm realizing I don't like the this function being named `tryToConcat...`, as it doesn't actually do the concat itself, it just tries to find the concat dimension. Maybe it can be `findConcatDim...` or something similar?

Reply:
sounds right!

Reply:
changed to findConcatDimForSameTrailingAndLeadingDims